pkg/trace/api/otlp.go

```diff
@@ -318,9 +310,21 @@ func (o *OTLPReceiver) ReceiveResourceSpans(rspans ptrace.ResourceSpans, header
 	if env == "" {
 		env = o.conf.DefaultEnv
 	}
-	if !hostok {
+
+	// Get the hostname or set to empty
+	// if source is empty
+	var hostname string
+	if srcok {
+		switch src.Kind {
+		case source.HostnameKind:
+			hostname = src.Identifier
+		default:
+			hostname = ""
+		}
+	} else {
 		hostname = o.conf.Hostname
 	}
+
```

Comment:
Can you please remove this line? I know it's nitty but trying to follow https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#general-guidelines (see point 3).